packages/beacon-node/src/network/network.ts

```diff
@@ -501,17 +501,30 @@ export class Network implements INetwork {
     peerId: PeerIdStr,
     request: deneb.BlobSidecarsByRangeRequest
   ): Promise<deneb.BlobSidecar[]> {
+    const fork = this.config.getForkName(this.clock.currentSlot);
```

Comment:
need to check if it's correct to use clock slot here, what if we range sync and wanna get deneb blob sidecars but clock slot is already electra?

Reply:
Good catch. I think we should use `request.startSlot` here. And if the request start slot is in deneb but the end slot is in electra, we still want to send out V1.